homeassistant/components/yamaha/media_player.py

```diff
@@ -143,46 +127,86 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Yamaha platform."""
-    # Keep track of configured receivers so that we don't end up
-    # discovering a receiver dynamically that we have static config
-    # for. Map each device from its zone_id .
-    known_zones = hass.data.setdefault(DOMAIN, {KNOWN_ZONES: set()})[KNOWN_ZONES]
-    _LOGGER.debug("Known receiver zones: %s", known_zones)
-
-    # Get the Infos for configuration from config (YAML) or Discovery
-    config_info = YamahaConfigInfo(config=config, discovery_info=discovery_info)
-    # Async check if the Receivers are there in the network
-    try:
-        zone_ctrls = await hass.async_add_executor_job(_discovery, config_info)
-    except requests.exceptions.ConnectionError as ex:
-        raise PlatformNotReady(f"Issue while connecting to {config_info.name}") from ex
-
-    entities = []
-    for zctrl in zone_ctrls:
-        _LOGGER.debug("Receiver zone: %s serial %s", zctrl.zone, zctrl.serial_number)
-        if config_info.zone_ignore and zctrl.zone in config_info.zone_ignore:
-            _LOGGER.debug("Ignore receiver zone: %s %s", config_info.name, zctrl.zone)
-            continue
-
-        assert config_info.name
-        entity = YamahaDeviceZone(
-            config_info.name,
-            zctrl,
-            config_info.source_ignore,
-            config_info.source_names,
-            config_info.zone_names,
+    host = config.get(CONF_HOST)
+    results = []
+    if host is not None:
+        _LOGGER.debug("Importing yaml single: %s", host)
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
         )
-
-        # Only add device if it's not already added
-        if entity.zone_id not in known_zones:
-            known_zones.add(entity.zone_id)
-            entities.append(entity)
-        else:
-            _LOGGER.debug(
-                "Ignoring duplicate zone: %s %s", config_info.name, zctrl.zone
+        results.append((host, result))
+    else:
+        ssdp_entries: list[SsdpServiceInfo] = await ssdp.async_get_discovery_info_by_st(
+            hass, "upnp:rootdevice"
+        )
+        matches = [
+            w
+            for w in ssdp_entries
+            if w.ssdp_location
+            and await YamahaConfigInfo.check_yamaha_ssdp(w.ssdp_location, hass)
+        ]
+        for entry in matches:
+            host = entry.ssdp_headers.get("_host")
```

Comment:
This was using `rxv.find` before. Why are we changing it?

Reply:
Because `rxv.find` does not catch certain exceptions, see https://github.com/wuub/rxv/issues/93 (and my PR there has been open for ages and the library is unmaintained).
So that's why I'm relying on the HA implementation here.

Reply:
This is only used for migration. So I assume the only people who use it, are precisely those for whom it works correctly.
We want to avoid a situation, where the old code detected the user's device, but the new one (for whatever reason) does not.

Reply:
Sure, I do want stress that I've made sure the code logic was the same by reusing `rvx_details` (https://github.com/wuub/rxv/blob/main/rxv/ssdp.py#L72).

Truth is that the `rvx.find` is unpredictable, depending on the number of devices on your network and the reply that comes back from the SSDP multicast, the `reponses` loop may bail out at any time.

So yes, I do agree that the people who use it are the ones that will use it in the migration, but the migration may fail randomly as well for them (and I'm probably one of the only ones having this issue due to a router sending back a weird Location header).

The current code would in theory only detect more devices where the old code may break randomly within the loop.

So just to reconfirm, you still want me to revert to rvx.find`?

Reply:
If you are confident, that this is the best way to guarantee a successful migration for people who were using this option, then it can stay the way it is.
I don't own the device, so it is hard for me to make a judgement call, how buggy the previously used library calls are.

Reply:
So more use of rxv.find was tried before so we could find a serial for a device
uuid, It really doesn't work well and caused crashes instability, it also
turns out not to be very good at finding devices. SSDP discovery is what it
is doing which IMHO HA does better.

Reply:
Thanks, I'll leave it like this for the moment, I think it should be more robust than the rxv implementation.

To be honest since @pssc has mentioned there was another integration on HACS (https://github.com/mvdwetering/yamaha_ynca), I switched over to using that. I haven't read much of the source code yet of https://github.com/mvdwetering/ynca but it does a better job at returning the actually renamed inputs from the device itself.

So in the long run maybe replacing YNC with YNCA might be a better implementation (but then again not all Yamaha receivers support YNCA), while keeping SSDP for the discovery part.

Reply:
@slyoldfox its also worth noting that only one device can connect with the YNCA protocol  certainly with my device, I think at some point the longer use of rxv need to be sorted either my forking or replacement